src/corner_kick/config/webpack.base.config.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * Specifies the webpack configuration for development
+ * and production. Provides support for Typescript and CSS files.
+ */
+
+const path = require('path');
+const webpack = require('webpack');
+const HtmlWebPackPlugin = require('html-webpack-plugin');
+const FriendlyErrorPlugin = require('friendly-errors-webpack-plugin');
+
+/**
+ * @description Webpack build.
+ */
+const generalWebpackBuild = {
+    stats: false,
+    // We support js, typescript, html, css, and image files.
+    // To add additional file support, add the required loader here.
+    module: {
+        rules: [
+            {
+                test: /\.tsx?$/,
+                exclude: /node_modules/,
+                use: {
+                    loader: 'ts-loader',
+                },
+            },
+            {
+                test: /\.html$/,
+                use: [
+                    {
+                        loader: 'html-loader',
+                        options: { minimize: false },
+                    },
+                ],
+            },
+            {
+                test: /\.css$/,
+                use: ['style-loader', 'css-loader'],
+            },
+            {
+                test: /\.(png|jpg|gif|svg)$/,
+                use: [
+                    {
+                        loader: 'file-loader',
+                        options: {},
+                    },
+                ],
+            },
+        ],
+    },
+    resolve: {
+        // Make sure if you add a new file support to add its extension here.
+        extensions: ['.js', '.ts', '.tsx', '.json'],
+
+        // We set an alias to our src directory to reduce the need for relative paths.
+        alias: {
+            SRC: path.resolve(__dirname, '../src'),
+        },
+    },
+};
+
+// Client build configuration
+module.exports = {
+    ...generalWebpackBuild,
+    // Our project entry point.
+    entry: path.resolve(__dirname, '../src/index.ts'),
+
+    // We generate a bundle in the build folder
+    output: {
+        path: path.resolve(__dirname, '../build'),
+        filename: '[name].[contenthash].js',
+    },
+
+    // This target allows us to access DOM libraries simultaneously
```

Comment:
Why do we wish to do this? (I can guess at performance, but generally comments should explicitly state _why_, rather then _what_ or _how_)

Reply:
Added more clarification. We could make the bundle smaller through various performance fix, but
would make the build file more complex. Because we run locally, we do not have to worry too much about bundle size.

```
// Webpack complains that our bundle is too large.
// We run the visualizer locally so we can keep our bundles bigger
// than the default max size.
```